### DISCUSSION TITLE
JS error :: SyntaxError: Identifier 'rawRequest' has already been declared

### DISCUSSION THREAD

**Discussion by sandipansaha** [breezefront/community#318] [1. Help]
Hi @vovayatsyuk 

I am seeing this error some custom pages. Even on some page where even no custom JS is added.

To see the error open the list page - URL
and then click on the event to reach the page - URL

Turbo mode is enabled and error seems to be related to that. But I am unable to figure out what could be the issue. If you kindly could guide me. I appreciate your help very much.

SyntaxError: Identifier 'rawRequest' has already been declared
    at o.assignNewBody (turbolinks-5.2.0.min.js:5:19830)
    at o.replaceBody (turbolinks-5.2.0.min.js:5:17960)
    at turbolinks-5.2.0.min.js:5:17532
    at e.renderView (turbolinks-5.2.0.min.js:5:16399)
    at o.render (turbolinks-5.2.0.min.js:5:17482)
    at e.render (turbolinks-5.2.0.min.js:5:16301)
    at t.renderSnapshot (turbolinks-5.2.0.min.js:5:22828)
    at t.render (turbolinks-5.2.0.min.js:5:22546)
    at r.render (turbolinks-5.2.0.min.js:6:755)
    at r.<anonymous> (turbolinks-5.2.0.min.js:5:27218)

![image](https://github.com/user-attachments/assets/fe5561fb-194f-48de-aad5-61424bf23591)

Thanks

**Comment by vovayatsyuk** [comment]
It's a inline LiveSearch code that uses `const`. Unfortunately, it's not compatible with Turbo:

```js
    const rawRequest = "";
    const searchRequest = rawRequest ? JSON.parse(rawRequest) : null;

    const rawResponse = "";
    const searchResponse = rawResponse ? JSON.parse(rawResponse) : null;

    const rawExecutionTime = "0";
    const executionTime = rawExecutionTime ? parseFloat(rawExecutionTime) : null;

    const requestId = "28f7f4f3\u002D3334\u002D42ea\u002Da3a1\u002D1b6c117e4ee3";

    const rawAttributes = "";
    const attributes = rawAttributes ? JSON.parse(rawAttributes) : null;

    const rawPageSize = "36";
    const pageSize = rawPageSize ? parseInt(rawPageSize) : null;

    const customerGroup = "b6589fc6ab0dc82cf12099d1c2d40ab994e8410c";

    window.LiveSearchMetrics = {
        searchRequest,
        searchResponse,
        requestId,
        executionTime,
        attributes,
        pageSize,
        context: {
            'customerGroup' : customerGroup
        }
    };

...
```

**Comment by vovayatsyuk** [comment]
I was thinking about Turbo lately. I tend to remove it in one of the next releases in favor of [view-transitions](https://developer.chrome.com/docs/web-platform/view-transitions). What do you think about such replacement?

**Reply by sandipansaha** [reply]
It is an interesting option. The example of Cross-document view transitions for MPA looks good. However, it does not support by the browsers widely like Safari and Mozilla, that could be a concern.
Also, with turbo mode I do not see much of if use anymore with speculation API implementation.

Thanks
